### PR TITLE
Function call cookbook 

### PR DESCRIPTION
**Pull request by msingh-openai**
## Summary

Update function call cookbook with latest model and API specifications 

## Motivation

The function call cook book updated to newer model 'GPT-4o' and used the latest API specification to invoke the Chat Completions API with "role":"tool" (rather than using "role":"function")

### CODE REVIEW

File: examples/How_to_call_functions_with_chat_models.ipynb [hunk 1]
```diff
@@ -33,37 +33,42 @@
...
     "from tenacity import retry, wait_random_exponential, stop_after_attempt\n",
     "from termcolor import colored  \n",
     "\n",
-    "GPT_MODEL = \"gpt-3.5-turbo-0613\"\n",
+    "GPT_MODEL = \"gpt-4o\"\n",
     "client = OpenAI()"
-   ]
+   ],
+   "outputs": [],
+   "execution_count": 62
```
Comment (teomusatoiu) [inline examples/How_to_call_functions_with_chat_models.ipynb:71]: can you re-run all cells so they're nicely numbered starting from 1?

File: examples/How_to_call_functions_with_chat_models.ipynb [hunk 2]
```diff
@@ -652,103 +732,130 @@
...
-   "execution_count": 19,
-   "id": "38c55083",
    "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "##### Steps to invoke a function call using Chat Completions API: \n",
+    "\n",
+    "Step #1: Prompt the model with content that may result in model selecting a tool to use. The description of the tools such as a function names and signature is defined in the 'Tools' list and passed to the model in API call. If selected, the function name and parameters are included in the response.  \n",
+    "Step #2:  Check programmatically if model wanted to call a function.  \n",
+    "Step #3: . Extract the function name and parameters from response, call the function with parameters. Append the result to messages.    \n",
```
Comment (teomusatoiu) [inline examples/How_to_call_functions_with_chat_models.ipynb:762]: there's an extra dot here

File: examples/How_to_call_functions_with_chat_models.ipynb [hunk 3]
```diff
@@ -652,103 +732,130 @@
...
+   "execution_count": 81
   },
   {
-   "cell_type": "code",
-   "execution_count": 19,
-   "id": "38c55083",
    "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "##### Steps to invoke a function call using Chat Completions API: \n",
```
Comment (teomusatoiu) [inline examples/How_to_call_functions_with_chat_models.ipynb:758]: Nit but could you make this a little bit more readable? Say add some bold for Step 1, Step 2 etc. I think you can also remove the '#' signs.

**Review by teomusatoiu — CHANGES_REQUESTED** [review]
Looks good! thanks for the contribution! I made a few small comments but once they're resolved it's good to ship!

**Comment by msingh-openai** [comment]
All comments addressed! Thanks for reviewing :)

**Review by teomusatoiu — APPROVED** [review]
Looks good! Approved!

**Pull request merged**